bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -476,6 +477,9 @@ WebViewEnvironment createEnvironment() {
 	String dataDir = System.getProperty(DATA_DIR_PROP);
 	String browserArgs = System.getProperty(BROWSER_ARGS_PROP);
 	String language = System.getProperty(LANGUAGE_PROP);
+	if (dataDir == null) {
+		dataDir = (String) browser.getDisplay().getData(DISPLAY_USER_DATA_FOLDER);
+	}
```

Comment:
Extract the logic to calculate `dataDir` to its own method: `getDataDir(...)`.

Also, why use both `browser.getDisplay()` and `Display.getCurrent()` as the displays when fetching the data? (see lines below too)

Reply:
Fixed with 6509979.